[kddockwidgets] seems to have optional deps

```
DragController.cpp.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) class std::shared_ptr<class spdlog::logger> __cdecl spdlog::get(char const *)" (__imp_?get@spdlog@@YA?AV?$shared_ptr@Vlogger@spdlog@@@std@@PEBD@Z)
  Hint on symbols that are defined and could potentially match:
    "__declspec(dllimport) public: static class QQmlEnginePrivate * __cdecl QQmlEnginePrivate::get(class QQmlEngine *)" (__imp_?get@QQmlEnginePrivate@@SAPEAV1@PEAVQQmlEngine@@@Z)
    "__declspec(dllimport) public: static class QQuickItemPrivate * __cdecl QQuickItemPrivate::get(class QQuickItem *)" (__imp_?get@QQuickItemPrivate@@SAPEAV1@PEAVQQuickItem@@@Z)
    "__declspec(dllimport) public: static class QQuickItemPrivate const * __cdecl QQuickItemPrivate::get(class QQuickItem const *)" (__imp_?get@QQuickItemPrivate@@SAPEBV1@PEBVQQuickItem@@@Z)

bin\kddockwidgets-qt62d.dll : fatal error LNK1120: 1 unresolved externals
```

Seems to lookup `qtdeclarative/qtquick` but is only declaring a dependency on `qtbase`

--- a/ports/kddockwidgets/portfile.cmake
+++ b/ports/kddockwidgets/portfile.cmake
@@ -2,7 +2,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO KDAB/KDDockWidgets
     REF "v${VERSION}" 
-    SHA512 4dccf24e901ab58d645478bc62ff9e72224dc11c3f39c53f5be5b188ece1bf8c682d50a42ece7a38400adfeb6147336795fcb86e903fd0957949c83f852c9b53 
+    SHA512 7b88f354e2aca4ac4c0f59874b6a7d6baaf77f5b54dd57b981ec7831e40acc0e2f6d3c6300af3d93c594bf34c7072c6a8a19a50c65039ccae22a9e47b90499d8
     HEAD_REF master
 )
 
@@ -21,9 +21,14 @@ vcpkg_cmake_configure(
     OPTIONS
         ${_qarg_OPTIONS}
         -DKDDockWidgets_QT6=ON
+        "-DKDDockWidgets_FRONTENDS=qtwidgets"
         -DKDDockWidgets_STATIC=${KD_STATIC}
         -DKDDockWidgets_PYTHON_BINDINGS=OFF
+        -DKDDockWidgets_TESTS=OFF
         -DKDDockWidgets_EXAMPLES=OFF
+        # https://github.com/KDAB/KDDockWidgets/blob/v2.1.0/CMakeLists.txt#L301
+        -DCMAKE_DISABLE_FIND_PACKAGE_spdlog=ON
+        -DCMAKE_DISABLE_FIND_PACKAGE_fmt=ON
 )
 
 vcpkg_cmake_install()

--- a/ports/kddockwidgets/vcpkg.json
+++ b/ports/kddockwidgets/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "kddockwidgets",
-  "version": "2.0.0",
+  "version": "2.1.0",
   "description": "KDAB's Dock Widget Framework for Qt",
   "homepage": "https://www.kdab.com/development-resources/qt-tools/kddockwidgets/",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3885,7 +3885,7 @@
       "port-version": 0
     },
     "kddockwidgets": {
-      "baseline": "2.0.0",
+      "baseline": "2.1.0",
       "port-version": 0
     },
     "kdsoap": {

--- a/versions/k-/kddockwidgets.json
+++ b/versions/k-/kddockwidgets.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d9644c9741f94c097fba186a4111f1dfcd68063e",
+      "version": "2.1.0",
+      "port-version": 0
+    },
+    {
       "git-tree": "5dcce3389ff415798302c843980052dd03314ddc",
       "version": "2.0.0",
       "port-version": 0

